listenbrainz/webserver/views/art_api.py

```diff
@@ -336,26 +349,18 @@ def cover_art_yim_2022(user_name):
     if image is None:
         raise APIBadRequest("Type of Image needs to be specified should be one of (stats, artists, albums, tracks, discovery-playlist, missed-playlist)")
 
-    if image not in ("stats", "artists", "albums", "tracks", "discovery-playlist", "missed-playlist"):
-        raise APIBadRequest(f"Invalid image type {image}. Image type should be one of (stats, artists, albums, tracks, discovery-playlist, missed-playlist)")
-
     stats = db_yim.get(user["id"], 2022)
     if stats is None:
         raise APIBadRequest(f"Year In Music report for user {user_name} not found")
 
-    if image == "stats":
-        svg = _cover_art_yim_stats(user_name, stats)
-    elif image == "albums":
-        svg = _cover_art_yim_albums(user_name, stats)
-    elif image == "tracks":
-        svg = _cover_art_yim_tracks(user_name, stats)
-    elif image == "artists":
-        svg = _cover_art_yim_artists(user_name, stats)
-    elif image == "discovery-playlist":
-        svg = _cover_art_yim_playlist(user_name, stats, "playlist-top-discoveries-for-year")
-    elif image == "missed-playlist":
-        svg = _cover_art_yim_playlist(user_name, stats, "playlist-top-missed-recordings-for-year")
-    else:
-        svg = None
-
-    return svg, 200, {"Content-Type": "image/svg+xml"}
+    match image:
+        case "stats": svg = _cover_art_yim_stats(user_name, stats)
+        case "albums": svg = _cover_art_yim_albums(user_name, stats)
+        case "tracks": svg = _cover_art_yim_tracks(user_name, stats)
+        case "artists": svg = _cover_art_yim_artists(user_name, stats)
+        case "discovery-playlist": svg = _cover_art_yim_playlist(user_name, stats, "playlist-top-discoveries-for-year")
+        case "missed-playlist": svg = _cover_art_yim_playlist(user_name, stats, "playlist-top-missed-recordings-for-year")
+        case other: raise APIBadRequest(f"Invalid image type {other}. Image type should be one of (stats, artists, albums, tracks, discovery-playlist, missed-playlist)")
+
+    status_code = 200 if svg is not None else 204
```

Comment:
is this necessary? since it'll be caught by the `case other`, right? This is something I've not used before, so I don't know if this is the best way to do things...

Reply:
This is for the case where the image type is correct but we didn't have enough data in the stats to generate a svg. The case other is for when the image type in the request was invalid.